bronchip/src/lib.rs

```diff
@@ -0,0 +1,51 @@
+use js_sys::Promise;
+use serde::{Deserialize, Serialize};
+use wasm_bindgen::prelude::*;
+use wasm_bindgen::JsValue;
+use wasm_bindgen_futures::{future_to_promise, JsFuture};
+use web_sys::{ReadableStream, ReadableStreamDefaultController, ReadableStreamDefaultReader};
+
+#[derive(Serialize, Deserialize)]
+pub struct ReadData {
+    value: Vec<u8>,
+    done: bool,
+}
+
+#[wasm_bindgen]
+pub struct StreamSource {
+    stream: ReadableStream,
+}
+
+#[wasm_bindgen]
+impl StreamSource {
+    #[wasm_bindgen(getter, js_name = type)]
+    pub fn type_(&self) -> JsValue {
+        JsValue::undefined()
```

Comment:
This is a "hack" to make `StreamSource` satisfy the `UnderlyingDefaultSource` interface in TS.